File: lib/features/sanction_management/utils/date_utils.dart
DateTime addWorkingDays(DateTime start, int days) {
  var date = start;
  var added = 0;

  while (added < days) {
    date = date.add(const Duration(days: 1));
    if (date.weekday <= DateTime.friday) {
      added++;
    }
  }
  return date;
}

File: backend/src/utils/date.utils.js
function addWorkingDays(startDate, days) {
  const date = new Date(startDate);
  let added = 0;

  while (added < days) {
    date.setDate(date.getDate() + 1);
    const day = date.getDay();
    if (day !== 0 && day !== 6) {
      added++;
    }
  }
  return date;
}

module.exports = { addWorkingDays };
